### PR TITLE
Plane_3-Plane_lt-map-patch

### PR DESCRIPTION
**Pull request by gunshi**
Patch for Plane_lt to recognise proportional planes as the same plane.

### CODE REVIEW

**Comment by sloriot** [comment]
Could you update your PR to only have one commit and at the same time amend it so that the commit message is more informative about the nature of the change?

**Comment by gunshi** [comment]
Updated.

**Comment by sloriot** [comment]
I think you should handle cases where `h2` might have `a()`, `b()`, `c()` or `d()` returning 0.

**Comment by gunshi** [comment]
Yes, sorry, forgot that. Handled now.

**Comment by sloriot** [comment]
Merged in CGAL-4.8-Ic-85

**Comment by sloriot** [comment]
`Nef_3/test/Nef_3/test_with_extended_homogeneous.cpp` fails at runtime on all platforms

**Comment by lrineau** [comment]
@gunshi, can you please comment on the failing tests?

**Comment by gunshi** [comment]
Yes when I run test_with_extended_homogenous.cpp, I get multiple assertion fails of the kind : assert(N1.is_valid(0,0)), the first of which occurs in the function call for simplification_SM() line 992.
It's taking me time to get to the root of it, should I close the request for the meanwhile?

**Comment by lrineau** [comment]
We can keep the PR open until you have another fix.

**Comment by gunshi** [comment]
Okay, so the error occurs when I compute the ratios of the a,b,c and d coefficients and store them in Kernel::RT which seems to store only the integer part of the ratio.
 When I print out the ratios for two planes A,B defined by
 A: 0 1,   0 0,  0 0,  0 -2
 B: 0 1,   0 0,  0 0,  0 -3
 the ratios I get are: 0 1, 0 1, 0 1, 0 1 ( I get 1.5 for the last ratio when i convert everything to double, but I wanted to avoid doing that due to precision loss ). Is there any way I could get around it?

File: Nef_3/include/CGAL/Nef_3/SNC_external_structure.h [hunk 1]
```diff
@@ -123,21 +123,53 @@ int sign_of(const CGAL::Plane_3<R>& h)
...
-    if ( (diff) != 0 ) return CGAL_NTS sign(diff) < 0;
-    diff = h1.d()-h2.d(); return CGAL_NTS sign(diff) < 0;
-  }
+    template <typename R>
+    bool operator()(const CGAL::Plane_3<R>& h1,
+                    const CGAL::Plane_3<R>& h2) const
+    {
+        typedef typename R::RT     RT;
+        bool flag=false;
+        RT ratioa,ratiob,ratioc,ratiod;
```
Comment (lrineau) [inline Nef_3/include/CGAL/Nef_3/SNC_external_structure.h]: `RT` means ring type. There is no division.

Use the following:
```C++
typedef typename R::FT     FT;
FT ratioa,ratiob,ratioc,ratiod;
```
it should work.
For the variable `RT diff`, [line 159][l159], I think you can stay with `RT`.

**Comment by gunshi** [comment]
Changes done, thanks!

**Comment by lrineau** [comment]
@gunshi Just to be sure, have you checked that the compilation and run of `test_with_extended_homogenous.cpp` are OK, now?

**Comment by lrineau** [comment]
@sloriot I think that patch may go to CGAL-4.8, as a bug-fix.

**Comment by gunshi** [comment]
Yes, i've checked they're okay.

**Comment by sloriot** [comment]
Merged in CGAL-4.8-Ic-137

**Pull request merged**